tests/integration/self_update_test.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -g expected_output
```

Comment:
Do we need this global variable? FWIU, `expected_output` is only used inside `self_update_and_check_version`, so can't we use a local variable for this?